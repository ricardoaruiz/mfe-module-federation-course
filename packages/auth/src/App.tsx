import React from 'react'

const App: React.FC = () => {
  return <h1>Auth works fine!!</h1>
}

export default App
