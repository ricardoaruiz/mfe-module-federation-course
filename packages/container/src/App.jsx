import React from 'react'

const App = () => {
  return (
    <h1>Container works fine!! Now with React</h1>
  )
}

export default App