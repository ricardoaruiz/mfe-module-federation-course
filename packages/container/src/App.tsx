import React from 'react'

import { Header } from './components/Header'

const App: React.FC = () => {
  return (
    <>
      <Header />
      <h1>Container works fine!! Now with React and Typescript</h1>
      <h2>Deployed by GithubActions on AWS S3 and Cloudfront</h2>
    </>
  )
}

export default App
