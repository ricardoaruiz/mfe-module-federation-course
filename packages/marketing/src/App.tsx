import React from 'react'
import { Router, Routes, Route } from 'react-router-dom'

import { BrowserHistory, MemoryHistory } from 'history'

import { Home, Price } from './pages'

type AppProps = {
  history: BrowserHistory | MemoryHistory
}

const App: React.FC<AppProps> = ({ history }) => {
  const [routerState, setRouterState] = React.useState({
    action: history.action,
    location: history.location,
  })

  React.useLayoutEffect(() => {
    const unlistenHistory = history.listen(setRouterState)
    return () => unlistenHistory()
  }, [history])

  React.useEffect(() => () => console.log('desmontand...'), [])

  return (
    <Router
      location={routerState.location}
      navigationType={routerState.action}
      navigator={history}
    >
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="price" element={<Price />} />
      </Routes>
    </Router>
  )
}

export default App
