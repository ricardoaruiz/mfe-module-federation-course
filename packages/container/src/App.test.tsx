/* eslint-disable react/display-name */
/* eslint-disable testing-library/no-node-access */
import React from 'react'
import { render } from '@testing-library/react'

import App from './App'

jest.mock('./components/MarketingApp', () => () => {
  const MarketingApp = () => <div>Mocked MarketingApp</div>
  return <MarketingApp />
})

describe('App', () => {
  it('should render correctly', () => {
    const { container } = render(<App />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
