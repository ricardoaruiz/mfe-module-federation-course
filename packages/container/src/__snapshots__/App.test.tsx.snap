// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<header
  class="sc-bczRLJ gEBsPY"
>
  <div
    aria-label="MFes - Module Federation"
    class="sc-gsnTZi ceSsda"
    role="presentation"
  >
    MFes - Module Federation
  </div>
  <div
    class="sc-dkzDqf beQdYK"
  >
    <button
      class="sc-hKMtZM cyqgDw"
      type="button"
    >
      Login
    </button>
  </div>
</header>
`;
